src/api/source/mvt.js

```diff
@@ -68,79 +74,104 @@ export default class MVT extends Base {
         }
         const tile = new VectorTile(new Protobuf(arrayBuffer));
         const mvtLayer = tile.layers[Object.keys(tile.layers)[0]];
-
-        const { points, featureGeometries, properties, numFeatures } = this._decodeMVTLayer(mvtLayer, this._metadata, MVT_EXTENT);
-
-
+        const { geometries, properties, numFeatures } = this._decodeMVTLayer(mvtLayer, this._metadata, MVT_EXTENT);
         const rs = rsys.getRsysFromTile(x, y, z);
-        const dataframeGeometry = this._metadata.geomType == geometryTypes.POINT ? points : featureGeometries;
-        const dataframe = this._generateDataFrame(rs, dataframeGeometry, properties, numFeatures, this._metadata.geomType);
+        const dataframe = this._generateDataFrame(rs, geometries, properties, numFeatures, this._metadata.geomType);
+
         return dataframe;
     }
 
 
     _decodeMVTLayer(mvtLayer, metadata, mvt_extent) {
-        let points;
-        if (metadata.geomType == geometryTypes.POINT) {
-            points = new Float32Array(mvtLayer.length * 2);
+        if (!mvtLayer.length) {
+            return { properties: [], geometries: {}, numFeatures: 0 };
+        }
+        if (!metadata.geomType) {
+            metadata.geomType = this._autoDiscoverType(mvtLayer);
         }
-        let featureGeometries = [];
+        switch (metadata.geomType) {
+            case geometryTypes.POINT:
+                return this._decode(mvtLayer, metadata, mvt_extent, new Float32Array(mvtLayer.length * 2));
+            case geometryTypes.LINE:
+                return this._decode(mvtLayer, metadata, mvt_extent, [], decodeLines);
+            case geometryTypes.POLYGON:
+                return this._decode(mvtLayer, metadata, mvt_extent, [], decodePolygons);
+            default:
+                throw new Error('MVT: invalid geometry type');
+        }
+    }
+
+    _autoDiscoverType(mvtLayer) {
+        const type = mvtLayer.feature(0).type;
+        switch (type) {
+            case mvtDecoderGeomTypes.point:
```

Comment:
I think that now this is `MVT_TO_CARTO_TYPES`, right?

Reply:
yep but a switch case reveals the intention better 👀